src/Compilation.zig

```diff
@@ -429,7 +429,10 @@ pub const AllErrors = struct {
                             try stderr.writeByte('\n');
                             try stderr.writeByteNTimes(' ', src.column);
                             ttyconf.setColor(stderr, .Green);
-                            try stderr.writeAll("^\n");
+                            try stderr.writeByte('^');
+                            // TODO basic unicode code point monospace width
+                            try stderr.writeByteNTimes('~', src.span.end - src.span.start - 1);
```

Comment:
I assume you are talking about half-width vs full-width characters?
Here's what I did: https://github.com/r00ster91/Conversant/blob/877ae89bddb2dd0653d38617a8b355136df5fcb2/src/editor.zig#L28-L41
Checking for whether the UTF-8 representation of the character takes up >= 3 bytes works surprisingly well.

Reply:
I've also already solved it in my C compiler but didn't include it here yet but maybe I should've
https://github.com/Vexu/arocc/blob/23133aed979e69148b487138e2638e7f76ded3c4/src/Source.zig#L85-L109

This doesn't account for multi codepoint characters like emojis but that shouldn't be too hard either if desired.

Reply:
Ooh, cool! Your solution definitely looks a lot more advanced than mine. Do you think we might want to add this to `std.unicode`? There is definitely demand for it but I remember that `std.unicode` shouldn't have too much functionality. Do you think this would be an ok addition?

Reply:
I do think it could be added to `std.unicode` but I'll do it in a separate PR since it'll also touch `std.zig.Ast.tokenLocation` and  people will have stronger opinions about it that this change.